src/AiAdapter.js

```diff
@@ -496,15 +488,15 @@ var ticTacToe = ticTacToe || {};
    * Determines whether a square is empty.
    * @param {integer} square
    */
-  app.AiAdapter.prototype.squareIsEmpty = function (square) {
-    return (square === 0) ? true : false;
+  app.AiAdapter.prototype.isSquareEmpty = function (square, board) {
```

Comment:
a player has to determine if they can play in a certain square and make that play. The engine will determine if it is legal and send the updated board to the other player or if it is illegal then it will send the board back to the player to make a legal move.